lib/ruby_routes/route.rb

```
@@ -1,310 +1,421 @@
+      end
+      frozen_key = raw_key.freeze
+      if @param_key_slots[0][0].nil?
+        @param_key_slots[0] = [hash_val, frozen_key]
+      elsif @param_key_slots[1][0].nil?
+        @param_key_slots[1] = [hash_val, frozen_key]
+      else
+        @param_key_slots[0] = @param_key_slots[1]
+        @param_key_slots[1] = [hash_val, frozen_key]
+      end
+      frozen_key
+    end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `build_param_cache_key` is too high. [<7, 22, 7> 24.12/23]

(Metrics/AbcSize)